src/sqlacodegen/generators.py

```diff
@@ -1342,3 +1342,132 @@ def render_relationship(self, relationship: RelationshipAttribute) -> str:
         kwargs["metadata"] = f"{{{self.metadata_key!r}: {rendered}}}"
         rendered_field = render_callable("field", kwargs=kwargs)
         return f"{relationship.name}: {annotation} = {rendered_field}"
+
+
+class SQLModelGenerator(DeclarativeGenerator):
+    def __init__(
+        self,
+        metadata: MetaData,
+        bind: Connection | Engine,
+        options: Sequence[str],
+        *,
+        indentation: str = "    ",
+        base_class_name: str = "SQLModel",
+    ):
+        super().__init__(
+            metadata,
+            bind,
+            options,
+            indentation=indentation,
+            base_class_name=base_class_name,
+        )
+
+    def collect_imports(self, models: Iterable[Model]) -> None:
+        super(DeclarativeGenerator, self).collect_imports(models)
+        if any(isinstance(model, ModelClass) for model in models):
+            self.remove_literal_import("sqlalchemy", "MetaData")
+            self.add_literal_import("sqlmodel", "SQLModel")
+            self.add_literal_import("sqlmodel", "Field")
+
+    def collect_imports_for_model(self, model: Model) -> None:
+        super(DeclarativeGenerator, self).collect_imports_for_model(model)
+        if isinstance(model, ModelClass):
+            for column_attr in model.columns:
+                if column_attr.column.nullable:
+                    self.add_literal_import("typing", "Optional")
+                    break
+            if model.relationships:
```

Comment:
Please add a blank line after each control block when more code follows on the same indentation level (e.g. to distinguish if...else blocks from unrelated code).
```suggestion

            if model.relationships:
```